src/main/java/org/kjy5/Main.java

```diff
@@ -93,6 +102,8 @@ public static void main(String[] args) {
     var virtualRootMapping = new LinkedHashMap<Tree, Tree>();
     var ChildListVirtualNodesMapping = new LinkedHashMap<Tree, ChildListVirtualNodes>();
 
+    // Would it be better to include the string in the ContentTuple?  Why or why not?  Also, is a
```

Comment:
I cannot put this inside the ContentTuple (I tried before) because I later union ContentTuple's and this extra field will make two tuples that should've been unioned become unique to each other

Reply:
I was unclear.  I was thinking of putting it in the data structure, but not having it affect the behavior of `equals()` or `hashCode()`.  (In other words, it is not part of the abstract state.)  I think that would avoid the problem you pointed out.  This is a design to consider, but the separate map does also work.

Reply:
I would prefer this. Does that mean I would have to override `equals` and `hashCode` myself? I have been using `record` because they have nice implementations already.

Reply:
Yes, you would override the methods yourself.   Like you, I use `record` where possible.  I think you can still use `record` and override the methods.  Alternately, you could make a normal class and override the methods.